Objects/listobject.c

```diff
@@ -2649,6 +2676,20 @@ list___init___impl(PyListObject *self, PyObject *iterable)
         (void)_list_clear(self);
     }
     if (iterable != NULL) {
+        if (_PyObject_HasLen(iterable) && self->ob_item == NULL) {
+            Py_ssize_t iter_len = PyObject_Size(iterable);
+            if (iter_len == -1) {
+                if (PyErr_ExceptionMatches(PyExc_Exception)) {
```

Comment:
What tests will fail if not silence any exceptions at all?

Reply:
How can `_PyObject_HasLen()` return true for generators?

Reply:
Ah, it is not true generator, it is a weakref. What if silence just TypeError?

Reply:
See `PyObject_LengthHint()`.

Reply:
Done in 0a6c8ba